src/com/cft/contactmerge/io/XmlImporter.java

```diff
@@ -1,31 +1,132 @@
 package com.cft.contactmerge.io;
 
 import com.cft.contactmerge.Contact;
+
+import java.io.FileNotFoundException;
 import java.util.*;
+import java.io.*;
+import java.util.stream.*;
+import javax.xml.parsers.*;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.NamedNodeMap;
+import org.xml.sax.SAXException;
 
 public class XmlImporter implements IImporter, Iterable<Contact> {
 
-    public void Load(String filename)
+    private Map<String, String> columnMap;
```

Comment:
Correct me if I'm wrong, but if we are importing the entire contents of the file, shouldn't this be `private Map<String, ArrayList<String>> columnMap;`... in other words `Map<PropertyTitle, ListOfPropertyValues>`

Keeping in mind the values can contain empty strings, and will be included in the order used to find the correct record value.

Reply:
This is a map of columns to properties. It doesn't contain any data.  The data is returned via the iterator one contact at a time and is never stored (at least in this class) as a collection.

Reply:
Yeah, as I started to step through the rest of the code, I realized how this would be used. Looks great :)